### PR TITLE
Various async function improvements

### PR DESCRIPTION
**Pull request by sdroege**
See individual commit messages and https://github.com/gtk-rs/gir/issues/508

### CODE REVIEW

File: src/analysis/bounds.rs [hunk 1]
```diff
@@ -122,8 +122,8 @@ impl Bounds {
                         let out_parameters = find_out_parameters(env, &function);
                         let parameters = format_out_parameters(&out_parameters);
                         let error_type = find_error_type(env, &function);
-                        type_string = format!("Fn(Result<{}, {}>) + Send + Sync + 'static", parameters, error_type);
-                        trampoline_type = Some(type_string.clone());
+                        type_string = format!("FnOnce(Result<{}, {}>) + Send + 'static", parameters, error_type);
+                        trampoline_type = Some((type_string.clone(), *self.unused.front().unwrap()));
```
Comment (sdroege) [inline src/analysis/bounds.rs]: This part here is ugly. Any suggestions?

**Comment by sdroege** [comment]
@EPashkin @antoyo please review and complain, thanks :)

File: src/analysis/bounds.rs (hunk 1 above)
Comment (EPashkin) [reply src/analysis/bounds.rs]: Only see do `let bound_name = *self.unused.front().unwrap();` before
and rename to `trampoline_info`

**Comment by EPashkin** [comment]
All gtk-rs builds fine with this for Windows on both nightly and stable.
@sdroege Thanks again.

File: src/analysis/bounds.rs (hunk 1 above)
Comment (sdroege) [reply src/analysis/bounds.rs]: Done

**Comment by EPashkin** [comment]
I waiting for @antoyo just in case.

**Comment by sdroege** [comment]
Maybe let's merge it if he doesn't have time to also review? We can still fix it up later if needed

**Comment by antoyo** [comment]
@sdroege I'll do it today, so merge tomorrow in any case.

**Comment by EPashkin** [comment]
@sdroege Thanks

**Comment by antoyo** [comment]
I approve this PR (a bit too late, but anyway :) ).

**Pull request merged**